src/ring_theory/algebra_tower.lean

```diff
@@ -295,45 +140,6 @@ variables {R S A}
 variables [comm_semiring R] [semiring S] [add_comm_monoid A]
 variables [algebra R S] [semimodule S A] [semimodule R A] [is_scalar_tower R S A]
 
-namespace submodule
-
-open is_scalar_tower
-
-theorem smul_mem_span_smul_of_mem {s : set S} {t : set A} {k : S} (hks : k ∈ span R s)
-  {x : A} (hx : x ∈ t) : k • x ∈ span R (s • t) :=
-span_induction hks (λ c hc, subset_span $ set.mem_smul.2 ⟨c, x, hc, hx, rfl⟩)
-  (by { rw zero_smul, exact zero_mem _ })
-  (λ c₁ c₂ ih₁ ih₂, by { rw add_smul, exact add_mem _ ih₁ ih₂ })
-  (λ b c hc, by { rw is_scalar_tower.smul_assoc, exact smul_mem _ _ hc })
-
-theorem smul_mem_span_smul {s : set S} (hs : span R s = ⊤) {t : set A} {k : S}
-  {x : A} (hx : x ∈ span R t) :
-  k • x ∈ span R (s • t) :=
-span_induction hx (λ x hx, smul_mem_span_smul_of_mem (hs.symm ▸ mem_top) hx)
-  (by { rw smul_zero, exact zero_mem _ })
-  (λ x y ihx ihy, by { rw smul_add, exact add_mem _ ihx ihy })
-  (λ c x hx, smul_left_comm c k x ▸ smul_mem _ _ hx)
-
-theorem smul_mem_span_smul' {s : set S} (hs : span R s = ⊤) {t : set A} {k : S}
-  {x : A} (hx : x ∈ span R (s • t)) :
-  k • x ∈ span R (s • t) :=
-span_induction hx (λ x hx, let ⟨p, q, hp, hq, hpq⟩ := set.mem_smul.1 hx in
-    by { rw [← hpq, smul_smul], exact smul_mem_span_smul_of_mem (hs.symm ▸ mem_top) hq })
-  (by { rw smul_zero, exact zero_mem _ })
-  (λ x y ihx ihy, by { rw smul_add, exact add_mem _ ihx ihy })
-  (λ c x hx, smul_left_comm c k x ▸ smul_mem _ _ hx)
-
-theorem span_smul {s : set S} (hs : span R s = ⊤) (t : set A) :
-  span R (s • t) = (span S t).restrict_scalars R :=
-le_antisymm (span_le.2 $ λ x hx, let ⟨p, q, hps, hqt, hpqx⟩ := set.mem_smul.1 hx in
-  hpqx ▸ (span S t).smul_mem p (subset_span hqt)) $
-λ p hp, span_induction hp (λ x hx, one_smul S x ▸ smul_mem_span_smul hs (subset_span hx))
-  (zero_mem _)
-  (λ _ _, add_mem _)
-  (λ k x hx, smul_mem_span_smul' hs hx)
-
-end submodule
-
 end semiring
```

Comment:
This whole section is now empty. Are you leaving it here in case future PRs add lemmas here?

Reply:
Done.